### PR TITLE
Add special -parse-as-library handling for swift_binary

### PR DESCRIPTION
**Pull request by keith**
This matches the behavior implemented in https://github.com/apple/swift-package-manager/pull/3410/ for single file modules.

Fixes https://github.com/bazelbuild/rules_swift/issues/913

### CODE REVIEW

**Comment by keith** [comment]
I'm not sure if we should do this or not. But folks having to know to pass `-parse-as-library` isn't ideal either.

**Review by jpsim — APPROVED** [review]
Looks correct to me!

File: swift/internal/swift_binary_test.bzl [hunk 1]
```diff
@@ -140,6 +140,25 @@ def _configure_features_for_binary(
...
+
+    Args:
+        srcs: A list of source files to check for the presence of `main.swift`.
+
+    Returns:
+        A list of compiler flags to add to `copts`
+    """
+    use_parse_as_library = len(srcs) == 1 and \
+                           srcs[0].basename != "main.swift"
+    return ["-parse-as-library"] if use_parse_as_library else []
```
[the remark this answers is not in the document]
Comment (thii) [reply swift/internal/swift_binary_test.bzl:160]: Looks like the failures are related

[the remark this answers is not in the document]
Comment (keith) [reply swift/internal/swift_binary_test.bzl:160]: yea this breaks the previous use case if you had some random filename with top level code, i'd have to fix those

**Pull request merged**